.changeset/bright-seas-march.md

```diff
@@ -0,0 +1,7 @@
+---
+'@leafygreen-ui/confirmation-modal': minor
+---
+
+- Dropped the `isRequired` from the `buttonText` prop type. As of [version 5.1.0](https://github.com/mongodb/leafygreen-ui/blob/main/packages/confirmation-modal/CHANGELOG.md#510), `buttonText` was marked as a deprecated prop that is now optional.
+- Fixed existing confirm button disabled state logic. When modal is closed and reopened when interacting with cancel or confirm button, confirm button will conditionally reset to disabled state only when `requiredInputText` is provided.
```

Comment:
This sentence is a little hard to follow, `When modal is closed and reopened when interacting with cancel or confirm button, confirm button will conditionally reset to disabled state only when `requiredInputText` is provided.`

Maybe something like, "When the `Modal` is closed using the cancel or confirm button and reopened, the confirm button will ..."